config/routes.rb

```diff
@@ -1,14 +1,19 @@
 Rails.application.routes.draw do
   namespace :api do
-    namespace :v1 do
-      get 'posts/index'
-      post 'posts/create'
-      get '/show/:id', to: 'posts#show'
-      delete '/destroy/:id', to: 'posts#destroy'
+    namespace :v2 do
+      resources :posts, only: [:index, :create, :show, :destroy] do
+        resources :likes, only: [:index, :create, :show] do
+          collection do
+            delete 'unlike'
+            get 'liked'
+          end
+        end
+        resources :replies, only: [:index, :create, :show, :destroy] do
```

Comment:
Nice use of `resources` it definitely helps group routes and makes it easier to navigate.